src/base/bittorrent/private/speedmonitor.cpp

```diff
@@ -27,16 +27,24 @@
  * exception statement from your version.
  */
 
-#include <QList>
 #include "speedmonitor.h"
 
 void SpeedMonitor::addSample(const SpeedSample &sample)
 {
-    m_speedSamples.push_back(sample);
+    static_assert(MAX_SAMPLES > 0, "MAX_SAMPLES should > 0");
+
     m_sum += sample;
-    if (m_speedSamples.size() > MAX_SAMPLES) {
-        m_sum -= m_speedSamples.front();
-        m_speedSamples.pop_front();
+    if (m_speedSamples.size() >= MAX_SAMPLES) {
```

Comment:
In most cases direct dependency on boost shouldn't be a problem, but in this case I remeber that @onto tried to use `boost::circular_buffer` and for some reason he abandoned that idea.
@onto what was the reason for not using `boost::circular_buffer`?
For reference look at @4261. A search&find pointed me to this reason https://github.com/qbittorrent/qBittorrent/pull/4261#issuecomment-178770443